src/index.js

```diff
@@ -6,41 +6,42 @@ const tags = tagsW3.map(tag => tag.replace(/[<|>]/g, ""));
 
 // Wait for the DOM to be fully loaded
 document.addEventListener("DOMContentLoaded", function () {
-
   uploadTagsFromLocalStorage(); // list all tags in localStorage
   updateProgressBar();
-  // Find the form element and add a submit event listener
-  const form = document.querySelector("form");
-  form.addEventListener("submit", function (event) {
-    // on submit
-    event.preventDefault(); // Prevent the default form submission
-    submitVal(); // call our submitVal function
-  });
+});
+
+// Find the form element and add a submit event listener
+const form = document.querySelector("form");
+form.addEventListener("submit", function (event) {
+  // on submit
+  event.preventDefault(); // Prevent the default form submission
+  submitVal(); // call our submitVal function
+});
+
+function submitVal() {
+  let val = document.getElementById("input").value;
+  // remove all white spaces and non-alphabetical characters except . ! - 
+  let regex = /[^a-zA-Z0-9!.\-]/g;
+  // change input to lower case and apply regex
+  let sanitizedVal = val.toLowerCase().replace(regex, "");
+  let li = "<li>" + sanitizedVal + "</li>";
+  // check if the input value is in the tags array
+  let isFound = tags.some(tag => tag.toLowerCase() === sanitizedVal);
+  // convert the list to an array
+  const getList = JSON.parse(localStorage.getItem("myTags"));
+  let isDuplicate = false;
 
-  function submitVal() {
-    let val = document.getElementById("input").value;
-    // remove all white spaces and non-alphabetical characters except . ! - 
-    let regex = /[^a-zA-Z0-9!.\-]/g;
-    // change input to lower case and apply regex
-    let sanitizedVal = val.toLowerCase().replace(regex, "");
-    console.log("Sanitized Value:", sanitizedVal);
-    let li = "<li>" + sanitizedVal + "</li>";
-    // check if the input value is in the tags array
-    let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
-    // convert the list to an array
-    const getList = JSON.parse(localStorage.getItem("myTags"));
-    let checkDuplicate = false;
-    // ? checks if the list is not empty first, to make includes() work
-    if (getList?.includes(sanitizedVal)) {
-      checkDuplicate = true;
-    }
-    if (found && !checkDuplicate) {
-        document.getElementById("list").insertAdjacentHTML("beforeend", li);
-        saveTagToLocalStorage(sanitizedVal); // Save tag to localStorage
-    }
+  // ? checks if the list is not empty first, to make includes() work
+  if (getList?.includes(sanitizedVal)) {
+    isDuplicate = true;
+  }
+
+  if (isFound && !isDuplicate) {
+    document.getElementById("list").insertAdjacentHTML("beforeend", li);
+    saveTagToLocalStorage(sanitizedVal);
     document.getElementById("input").value = "";
   }
-});
+}
 
```

Comment:
This file looks so much better! I also like to have a space between certain chunks of code, it's so much better for readability!